crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -297,6 +297,97 @@ struct Inner {
     sequence_state: SequencesState,
 }
 
+impl Inner {
+    fn validate(&self) {
+        #[derive(Default)]
+        struct State {
```

Comment:
I've decided to pull the validation stuff out because I think I want a refactor to this whole inserted/deleted_tables state stuff anyway, in a follow up to all this.